Microsoft.Research/Contracts/WindowsBase/System.Windows.Threading.DispatcherObject.cs

```diff
@@ -22,7 +22,26 @@ public abstract class DispatcherObject
   {
     protected DispatcherObject() { }
 
-    public Dispatcher Dispatcher { get { Contract.Ensures(Contract.Result<Dispatcher>() != null); return null; } }
+    // DetachFromDispatcher is e.g. called by Freezable.Freeze()
+    //
+    // This method allows certain derived classes to break the dispatcher affinity
+    // of our objects.
+    // [FriendAccessAllowed] // Built into Base, also used by Framework.
+    internal void DetachFromDispatcher()
```

Comment:
It does not help the contracts, but to understand what's going on here any why the contract must be this way.